### PR TITLE
399 - Fetch single document from community API.

### PR DESCRIPTION
**Pull request by chris-faulkner-mojdt**
gets a single document from the community API. Followed the approach of sending a ResponseEntity from community API. In order to handle and forward HTTP headers, had to use "exchange" from teh WebClient API rather than "retrieve". This changes exception handling and also the normal "blockOptional" call gave JVM warnings about blocking the thread, hence use of CompletableFuture.

IN order to trim down the Offender Rest Client, I have moved responsibilities for "documents" to DocumentRestClient.

### CODE REVIEW

File: src/main/java/uk/gov/justice/probation/courtcaseservice/application/ApplicationExceptionHandler.java [hunk 1]
```diff
@@ -26,6 +27,14 @@
                 .userMessage(e.getMessage()).build(), NOT_FOUND);
     }
 
+    @ExceptionHandler(DocumentNotFoundException.class)
```
Comment (jhevans) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/application/ApplicationExceptionHandler.java:30]: I've implemented a generic `RestResourceNotFound` exception and handler as part of my changes as it happens: https://github.com/ministryofjustice/court-case-service/pull/79

If mine gets merged first could consider pulling from master and applying the same pattern and could get rid of this handler

File: src/main/java/uk/gov/justice/probation/courtcaseservice/controller/OffenderController.java [hunk 2]
```diff
@@ -59,4 +69,23 @@ ProbationRecord getProbationRecord(@ApiParam(name = "crn", value = "CRN for the
     RequirementsResponse getRequirements(@PathVariable String crn, @PathVariable String convictionId) {
         return new RequirementsResponse(offenderService.getConvictionRequirements(crn, convictionId));
     }
+
+    @ApiOperation(value = "Gets a document by ID for a CRN.")
+    @ApiResponses(
+        value = {
+            @ApiResponse(code = 200, message = "OK", response = RequirementsResponse.class),
```
Comment (jhevans) [inline src/main/java/uk/gov/justice/probation/courtcaseservice/controller/OffenderController.java:76]: Response type needs amending

Comment (chris-faulkner-mojdt) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/controller/OffenderController.java:76]: Done

File: src/main/java/uk/gov/justice/probation/courtcaseservice/application/ApplicationExceptionHandler.java (hunk 1 above)
Comment (chris-faulkner-mojdt) [reply src/main/java/uk/gov/justice/probation/courtcaseservice/application/ApplicationExceptionHandler.java:30]: Cheers - think it's going to be the other way around, will help merge

**Pull request merged**